File: java/BinaryGapLength.java
public class BinaryGapLength 
{
    public static void main( String[] args )
    {
        int myint = 9;
        String mybyte = Integer.toBinaryString(myint);
        System.out.println("Binary: " + mybyte);
        int highestbinarygaplength = 0;
        int totalbinarygap = 0;        
        boolean binarystartcount = false;
        for (int i = (mybyte.length()-1); i >= 0; i--) {            
            switch (String.valueOf(mybyte.charAt(i))) {
                case "0":
                    if (binarystartcount) {
                        totalbinarygap = totalbinarygap + 1;                        
                    }                    
                    break;
                case "1":               
                    if (totalbinarygap == 0) {
                        binarystartcount = true;
                    } else {
                        if (totalbinarygap > highestbinarygaplength) {
                            highestbinarygaplength = totalbinarygap;
                        }
                        totalbinarygap = 0;
                    }
                    break;
                default:
                    break;
            }
        }
        System.out.println("Highest Binary Gap Length: " +highestbinarygaplength);        
    }
}

File: ruby/byte/binarygaplength.rb
String myint = 1041.to_s(2)
p "Binary: #{myint}"
Integer i = myint.size
Integer highestbinarygaplength = 0
Integer totalbinarygap = 0
Integer startcount = 0
loop do
    if i >= 0            
        case myint[i]
        when "0"
            if startcount == 1            
                totalbinarygap = totalbinarygap + 1                
            end
        when "1"
            if totalbinarygap == 0
                startcount = 1
            else
                if totalbinarygap > highestbinarygaplength
                    highestbinarygaplength = totalbinarygap                    
                end
                totalbinarygap = 0
            end
        else
            highestbinarygaplength = highestbinarygaplength
        end
    else
        break   
    end
    i = i - 1
end
p "Total binarygap: #{highestbinarygaplength}" 